lib/ansible/modules/clustering/kubevirt/kubevirt_cdi_upload.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+
+module: kubevirt_cdi_upload
+
+short_description: Upload local VM images to CDI Upload Proxy.
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+
+description:
+  - Use Openshift Python SDK to create UploadTokenRequest objects.
+  - Transfer contents of local files to the CDI Upload Proxy.
+
+extends_documentation_fragment:
+  - k8s_auth_options
+
+options:
+  pvc_name:
+    description:
+      - Use to specify the name of the target PersistentVolumeClaim.
+    required: true
+  pvc_namespace:
+    description:
+      - Use to specify the namespace of the target PersistentVolumeClaim.
+    required: true
+  upload_host:
+    description:
+      - URL containing the host and port on which the CDI Upload Proxy is available.
+  upload_host_verify_ssl:
+    description:
+      - Whether or not to verify the CDI Upload Proxy's SSL certificates.
+    default: true
+    type: bool
+  path:
+    description:
+      - Path of local image file to transfer.
```

Comment:
Is this qcow/raw, or what formats are supported?

Reply:
Dunno. I guess it'd make most sense to reference CDI's documentation on this, but CDI does not document this in an explicit way. Their [example](https://github.com/kubevirt/containerized-data-importer/blob/master/doc/upload.md#upload-an-image) suggests qcow2 is supported, but that's all I know.

Reply:
Yeah, that's potentially a good idea. Though I'd need some clarification on what's actually supported first and that'll take a bit of time.